src/main/java/baritone/command/defaults/ToMapCommand.java

```diff
@@ -0,0 +1,340 @@
+        ctx.worldData().getCachedWorld().reloadAllFromDisk();
+        boolean everyChunkWasLoaded = true;
+        for (int[] regionCoords : getRegionCoordsOnDisk()) {
+            int rX = regionCoords[0];
+            int rZ = regionCoords[1];
+            CachedWorld cw = (CachedWorld) ctx.worldData().getCachedWorld(); // is this cast a fair assumption?
+            cw.tryLoadFromDisk(rX, rZ);
+            ICachedRegion cr = cw.getRegion(rX, rZ);
+            System.out.println("Region " + rX + " " + rZ + " was loaded: " + (cr != null));
+            everyChunkWasLoaded &= cr != null;
+        }
+        return everyChunkWasLoaded;
+    }
+
+
+    @Override
+    public void execute(String label, IArgConsumer args) throws CommandException {
+        WorldScanner.INSTANCE.repack(ctx);
+        ctx.worldData().getCachedWorld().save();
+//        ctx.worldData().getCachedWorld().reloadAllFromDisk();
+        System.out.println("Loading cached chunks");
+//        baritone.getWorldProvider().getCurrentWorld().getCachedWorld().reloadAllFromDisk();
+//        if(!isEveryChunkedLoaded()){
+//            return;
+//        }
+//        System.out.println("All cached chunks loaded succesfully");
+        for (int[] regionCoords : getRegionCoordsOnDisk()) {
+            int rX = regionCoords[0];
+            int rZ = regionCoords[1];
+            executor.execute(saveToImage(rX, rZ));
+//            saveToImage(rX, rZ).start();
+        }
+        executor.shutdown();
+        while(!executor.isTerminated());
+    }
+
+    private Runnable saveToImage(int rX, int rZ) {
+        return new Runnable() {
+            @Override
+            public void run() {
+                CachedWorld cw = (CachedWorld) ctx.worldData().getCachedWorld(); // is this cast a fair assumption?
+                cw.tryLoadFromDisk(rX, rZ);
+                ICachedRegion cr = cw.getRegion(rX, rZ);
+//            System.out.println("cr is null: " + (cr == null));
+//            if (cr == null)
+//                continue;
+                Map<String, int[]> cornerBlockCoordinatesForZeroZero = getBlockCoordsOfRegionCorners(rX, rZ);
+                int[] nw = cornerBlockCoordinatesForZeroZero.get("NW");
+                Map<Integer, Map<Integer, String>> topView = new LinkedHashMap<>(); // in block coordinates
+                Pattern p = Pattern.compile("Block\\{minecraft:(.*)\\}");
+                logDirect("Getting top blocks for region " + rX + " " + rZ);
+                for (int x = nw[0]; x < nw[0] + 512; x++) {
+                    if (!topView.containsKey(x)) {
+                        HashMap<Integer, String> zId = new LinkedHashMap<>();
+                        topView.put(x, zId);
+                    }
+                    for (int z = nw[1]; z < nw[1] + 512; z++) {
+                        for (int y = 256; y >= 0; y--) {
+                            try {
+                                BlockState bs = cr.getBlock(x, y, z);
```

Comment:
this uses coordinates relative to the region, as shown in `BlockStateInterface#get0`
Swallowing exceptions seems to be a bad idea, this one causes up to 512*512*256 OutOfBounds exceptions per region and no one noticed it. Now I'm getting actual null blockstates.

Reply:
I don't have the time to verify this now. Are you implying that there's a problem in baritone's code itself or is it this problem restricted to ToMapCommand? Also, do you think this is easily fixable?

Reply:
Actuall I don't know. This might be a problem of ToMapCommand not accessing the cache correctly or baritone not returning blockstates it is supposed to return. I will try to use the normal ctx.get access method now.